lib/pbench/test/unit/agent/task/test_tools_kill.py

```diff
@@ -141,8 +192,31 @@ def iterdir(self):
             and pairs[1][1] == "ghijkl"
         )
 
+    @staticmethod
+    def test_gen_host_names_no_tool_groups(monkeypatch):
+        """Verify kill.gen_host_names() is a no-op when we have no tool groups
+        present.
+        """
+
+        def mock_gen_tool_groups(pbench_run: str) -> Iterable[ToolGroup]:
+            """Override kill.gen_tool_groups definition to yield no ToolGroup
+            objects.
+            """
+            for hosts in []:
+                yield ToolGroup(hosts)
+
+        monkeypatch.setattr(
+            "pbench.cli.agent.commands.tools.kill.gen_tool_groups", mock_gen_tool_groups
+        )
+        hosts = list(kill.gen_host_names(pathlib.Path("no-tool-group-dirs")))
+        assert len(hosts) == 0
```

Comment:
Supposedly, it's "not Pythonic" to check if the length of a list is zero.  Instead, we're supposed to look at the list itself in a boolean context, e.g.:
```
        assert not list(kill.gen_host_names(pathlib.Path("no-tool-group-dirs")))
```
However, that's not very satisfying.  How about this, instead:
```
        assert not any(kill.gen_host_names(pathlib.Path("no-tool-group-dirs")))
```